go/vt/vitessdriver/driver.go

```diff
@@ -231,7 +251,88 @@ func (c *conn) Close() error {
 	return nil
 }
 
+// DistributedTxFromSessionToken allows users to send serialized sessions over the wire and
+// reconnect to an existing transaction. Setting the sessionToken and address on the
+// supplied configuration is the minimum required
+func DistributedTxFromSessionToken(ctx context.Context, c Configuration) (*sql.Tx, error) {
+	if c.SessionToken == "" {
+		return nil, errors.New("c.SessionToken is required")
+	}
+	if c.Address == "" {
```

Comment:
Is this to prevent clients from connecting to different vtgates ? In theory the transaction is help open at the vttablet layer and we should be able to connect through any vtgate to have the transaction continue to work

Reply:
The address isn't serialized as a part of the session object and so is required. I could remove this check and let it error further down the line

Reply:
check removed